aiida/tools/importexport/dbexport/__init__.py

```diff
@@ -138,241 +365,225 @@ def export(
-    :type include_logs: bool
+    # Turn sets into lists to be able to export them as JSON metadata.
+    for entity, entity_set in entities_starting_set.items():
+        entities_starting_set[entity] = list(entity_set)  # type: ignore
 
-    :param kwargs: graph traversal rules. See :const:`aiida.common.links.GraphTraversalRules` what rule names
-        are toggleable and what the defaults are.
+    metadata = {
+        'aiida_version': get_version(),
+        'export_version': EXPORT_VERSION,
+        'all_fields_info': all_fields_info,
+        'unique_identifiers': unique_identifiers,
+        'export_parameters': {
+            'graph_traversal_rules': traversal_rules,
+            'entities_starting_set': entities_starting_set,
+            'include_comments': include_comments,
+            'include_logs': include_logs,
+        },
+    }
 
-    :raises `~aiida.tools.importexport.common.exceptions.ArchiveExportError`: if there are any internal errors when
-        exporting.
-    :raises `~aiida.common.exceptions.LicensingException`: if any node is licensed under forbidden license.
-    """
-    from collections import defaultdict
-    from aiida.tools.graph.graph_traversers import get_nodes_export
+    all_node_uuids = {node_pk_2_uuid_mapping[_] for _ in node_ids_to_be_exported}
 
-    if silent:
-        logging.disable(level=logging.CRITICAL)
+    # we get the node data last, because it is generally the largest data source
+    # and later we may look to stream this data in chunks
+    node_data = _get_node_data(node_ids_to_be_exported, silent)
 
-    EXPORT_LOGGER.debug('STARTING EXPORT...')
+    close_progress_bar(leave=False)
 
-    # Backwards-compatibility
-    entities = deprecated_parameters(
-        old={
-            'name': 'what',
-            'value': kwargs.pop('what', None)
-        },
-        new={
-            'name': 'entities',
-            'value': entities
-        },
+    return ExportData(
+        metadata,
+        all_node_uuids,
+        groups_uuid,
+        links_uuid,
+        export_data,
+        node_data,
     )
 
-    type_check(entities, (list, tuple, set), msg='`entities` must be specified and given as a list of AiiDA entities')
-    entities = list(entities)
-
-    type_check(folder, (Folder, ZipFolder), msg='`folder` must be specified and given as an AiiDA Folder entity')
 
-    all_fields_info, unique_identifiers = get_all_fields_info()
+def _get_starting_node_ids(entities: List[Any], silent: bool) -> Tuple[DefaultDict[str, Set[str]], Set[int]]:
+    """Get the starting node UUIDs and PKs
 
-    entities_starting_set = defaultdict(set)
-
-    # The set that contains the nodes ids of the nodes that should be exported
-    given_node_entry_ids = set()
-    given_log_entry_ids = set()
-    given_comment_entry_ids = set()
+    :param entities: a list of entity instances
+    :param silent: suppress console prints and progress bar.
 
+    :raises exceptions.ArchiveExportError
```

Comment:
```suggestion
    :raises exceptions.ArchiveExportError:
```

Reply:
fixed in b689f5e6986c15ab18ad225e5659f1ef3bf8f9c0